src/module.cc

```diff
@@ -0,0 +1,338 @@
+  sigfillset(&sigmask);
+  pthread_sigmask(SIG_SETMASK, &sigmask, &sigmask);
+  pthread_t thread;
+  const int err = pthread_create(&thread, &attr, thread_main, nullptr);
+  pthread_sigmask(SIG_SETMASK, &sigmask, nullptr);
+  pthread_attr_destroy(&attr);
+  if (err != 0) {
+    fprintf(stderr, "nodereport: pthread_create: %s\n", strerror(err));
+    fflush(stderr);
+    return -err;
+  }
+  return 0;
+}
+#endif
+
+
+/*******************************************************************************
+ * Native module initializer function, called when the module is require'd
+ *
+ ******************************************************************************/
+void Initialize(v8::Local<v8::Object> exports) {
+  v8::Isolate* isolate = Isolate::GetCurrent();
+  node_isolate = isolate;
+
+  SetLoadTime();
+
+  const char* trigger_events = getenv("NODEREPORT_EVENTS");
+  if (trigger_events != NULL) {
+    nodereport_events = ProcessNodeReportEvents(trigger_events);
+  }
+  const char* core_dump_switch = getenv("NODEREPORT_COREDUMP");
+  if (core_dump_switch != NULL) {
+    nodereport_core = ProcessNodeReportCoreSwitch(core_dump_switch);
+  }
+  const char* trigger_signal = getenv("NODEREPORT_SIGNAL");
+  if (trigger_signal != NULL) {
+    nodereport_signal = ProcessNodeReportSignal(trigger_signal);
+  }
+  const char* report_name = getenv("NODEREPORT_FILENAME");
+  if (report_name != NULL) {
+    ProcessNodeReportFileName(report_name);
+  }
+  const char* directory_name = getenv("NODEREPORT_DIRECTORY");
+  if (directory_name != NULL) {
+    ProcessNodeReportDirectory(directory_name);
+  }
+  const char* verbose_switch = getenv("NODEREPORT_VERBOSE");
+  if (verbose_switch != NULL) {
+	  nodereport_verbose = ProcessNodeReportVerboseSwitch(verbose_switch);
+  }
+
+  // If NodeReport requested for fatalerror, set up the V8 call-back
+  if (nodereport_events & NR_FATALERROR) {
+    isolate->SetFatalErrorHandler(OnFatalError);
+  }
+
+  // If NodeReport requested for exceptions, tell V8 to capture stack trace and set up the callback
+  if (nodereport_events & NR_EXCEPTION) {
+    isolate->SetCaptureStackTraceForUncaughtExceptions(true, 32, v8::StackTrace::kDetailed);
+    // The hook for uncaught exception won't get called unless the --abort_on_uncaught_exception option is set
+    v8::V8::SetFlagsFromString("--abort_on_uncaught_exception", sizeof("--abort_on_uncaught_exception")-1);
+    isolate->SetAbortOnUncaughtExceptionCallback(OnUncaughtException);
+  }
+
+#ifndef _WIN32
+  // If NodeReport requested on external user signal set up watchdog thread and callbacks
+  if (nodereport_events & NR_SIGNAL) {
+    uv_sem_init(&report_semaphore, 0);
+    if (StartWatchdogThread(ReportSignalThreadMain) == 0) {
+      uv_async_init(uv_default_loop(), &nodereport_trigger_async, SignalDumpAsyncCallback);
+      uv_unref(reinterpret_cast<uv_handle_t*>(&nodereport_trigger_async));
+      RegisterSignalHandler(nodereport_signal, SignalDump, false);
+    }

```

Comment:
This block of code is duplicated here: https://github.com/nodejs/nodereport/pull/2/files#r77156266